src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -647,20 +647,60 @@ private static TypeDeclDesc getTheNearestClass(String outerTypeDeclName,
      * @param secondTypeDecl second input type declaration
      * @return difference between type declaration name matching count
      */
-    private static int getTypeDeclarationNameMatchingCountDiff(String outerTypeDeclName,
+    private static int getAnonTypeDeclarationNameMatchingCountDiff(String outerTypeDeclName,
                                                                TypeDeclDesc firstTypeDecl,
                                                                TypeDeclDesc secondTypeDecl) {
         int diff = Integer.compare(
-            CheckUtil.typeDeclarationNameMatchingCount(
+            getAnonSuperTypeMatchingCount(
                 outerTypeDeclName, secondTypeDecl.getQualifiedName()),
-            CheckUtil.typeDeclarationNameMatchingCount(
+            getAnonSuperTypeMatchingCount(
                 outerTypeDeclName, firstTypeDecl.getQualifiedName()));
         if (diff == 0) {
             diff = Integer.compare(firstTypeDecl.getDepth(), secondTypeDecl.getDepth());
         }
         return diff;
     }
```

Comment:
While we are here, we can also refactor this method a bit to use immutable data and make what we are doing more clear:

```java
    final int firstMatchCount = countMatchingQualifierChars(outerTypeName, firstType.getQualifiedName());
    final int secondMatchCount = countMatchingQualifierChars(outerTypeName, secondType.getQualifiedName());

    final int matchDistance = Integer.compare(secondMatchCount, firstMatchCount);

    final int result;
    if (matchDistance == 0) {
        result = Integer.compare( firstType.getDepth(),  secondType.getDepth());
    } else {
        result = matchDistance;
    }

    return result;
```